crypto/engine/eng_rdrand.c

```diff
@@ -20,28 +20,15 @@
      defined(__x86_64) || defined(__x86_64__) || \
      defined(_M_AMD64) || defined (_M_X64)) && defined(OPENSSL_CPUID_OBJ)
 
-size_t OPENSSL_ia32_rdrand(void);
+size_t OPENSSL_ia32_rdrand_bytes(unsigned char *buf, size_t len);
 
 static int get_random_bytes(unsigned char *buf, int num)
 {
-    size_t rnd;
-
-    while (num >= (int)sizeof(size_t)) {
-        if ((rnd = OPENSSL_ia32_rdrand()) == 0)
-            return 0;
-
-        *((size_t *)buf) = rnd;
-        buf += sizeof(size_t);
-        num -= sizeof(size_t);
-    }
-    if (num) {
-        if ((rnd = OPENSSL_ia32_rdrand()) == 0)
-            return 0;
-
-        memcpy(buf, &rnd, num);
+    if (num < 0 || (unsigned int)num > (size_t)-1) {
```

Comment:
The torture never stops, huh? Now --strict-warnings say "comparison is always false due to limited range of data type". And we do demand --strict-warngings build to pass [at least on Linux]. I tried to hint that I for one don't view the check as essential [by emphasizing "if" in "if one aims to cover the corner case"]. But otherwise one can check for `(unsigned int)num != (size_t)num`. Or one can do it implicitly by returning `(unsigned int)num == OPENSSL_ia32_rdrand_bytes(buf, (size_t)num)`.

[Just in cases, none of this will make need-cla pass.]